docs/src/format/table/index/vector/index.md

```diff
@@ -0,0 +1,286 @@
+
+## Overview
+
+Lance's vector indexing system is designed for efficient similarity search across large-scale vector datasets. Each index is composed of two main components:
+
+### Index Architecture
+
+Lance splits each index into 2 parts:
+
+- **Sub-index**: The index structure, which can be FLAT or HNSW.
+- **Quantization**: The quantization method, which can be PQ, SQ, or FLAT (no quantization).
+
+### Sub-Index Types
+
+The sub-index determines how vectors are organized for search:
+
+- **FLAT**: Exact search with no approximation - scans all vectors
+- **HNSW**: Hierarchical Navigable Small World graphs for fast approximate search
+
+### Quantization Methods
+
+The quantization method determines how vectors are stored and compressed:
+
+- **Product Quantization (PQ)**: Compresses vectors by splitting them into smaller sub-vectors and quantizing each independently
+- **Scalar Quantization (SQ)**: Applies scalar quantization to each dimension of the vector independently  
+- **FLAT**: No quantization, keeps original vectors for exact search
+
+### Common Index Combinations
+
+- **IVF_PQ**: Inverted File with Product Quantization - combines IVF clustering with PQ compression for efficient storage and search
+- **IVF_HNSW_SQ**: Inverted File with HNSW - uses IVF for coarse clustering and HNSW for fine-grained search with scalar quantization
+- **IVF_SQ**: Inverted File with Scalar Quantization - combines IVF clustering with scalar quantization for balanced compression
+- **IVF_FLAT**: Inverted File with FLAT - uses IVF clustering with exact vector storage for precise search within clusters
+
+## Vector Index Format
+
+Vector Index V3 is the latest version of the vector index format. Each vector index is stored as 2 regular Lance files:
+
+- `index.idx`: The index structure file containing the search graph/structure with index-specific schema
+- `auxiliary.idx`: The vector storage file containing (quantized) vectors with schema `| _rowid: u64 | vector: fixed_size_list(dimension) |`
+
+### IVF (Inverted File) Context
+
+Most Lance indices use IVF (Inverted File) as the primary clustering mechanism:
+
+- **IVF** partitions the dataset into clusters using k-means clustering
+- Each cluster (partition) contains vectors that are similar to the cluster centroid
+- During search, only the most relevant clusters are examined, dramatically reducing search time
+- IVF can be combined with any sub-index type (FLAT, HNSW) and quantization method (PQ, SQ, FLAT)
+
+## File Structure
+
+The Lance index format stores metadata and data across two files with specific schemas and metadata fields.
+
+### Index File Metadata
+
+The index file contains metadata in its schema metadata to describe the index configuration and structure:
+
+#### "lance:index"
+
+Contains basic index configuration information:
+
+```rust
+#[derive(Serialize, Deserialize, Debug)]
```

Comment:
as a format spec we should probably describe this in JSON schema, not in rust code